### ISSUE TITLE
/search displays wrong laser

### ISSUE THREAD

**Issue by mfnalex** [JEFF-Media-GbR/Spigot-InvUnloadPlus#31]
Currently, the laser shown by /search shows the last unload results instead of the chests where the items have been found.

**Comment by taupemoi** [comment]
You probably noticed this already, but just in case: the purple particles highlighting the chest itself work correctly - it's just the laser which is incorrect.

Also, if no /unload happened before the /search command, no laser shows up at all.

**Comment by mfnalex** [comment]
Yes, the case why the laser doesn't properly atm is because I switched back to old particle style laser because the new laser I used for a few versions was blocking the chest so It could not be opened. It will be fixed this week but probably not today

**Comment by taupemoi** [comment]
No worries, thanks for all your great work!

**Comment by Engineer1986** [comment]
Hi, could you have a look at this bug if this can be fixed? They have this plugin on the Datblock Minecraft server and i'm fighting with the issue a few times a week. I had a look into the code before i found this bugreport. I think the fix should be something like this?

replace this in commandSearchitem.java:
```
for(Block block : affectedChests) {
            main.visualizer.chestAnimation(block,p);

//if(main.getConfig().getBoolean("laser-animation")) { // On search, ALWAYS play the animation
                main.visualizer.play(p);
            //}
        }
```

With this:
```
for(Block block : affectedChests) {
            main.visualizer.chestAnimation(block,p);

        }
main.visualizer.play(p, affectedChests);
```

Thanks a lot!

**Comment by mfnalex** [comment]
Hi, thanks for the hint. Unfortunately I couldnt really reproduce the problem yet. Can you please compile the plugin and test if your fix worked? If yes, I could release it today

**Comment by Engineer1986** [comment]
Hi, i had a try to compile the plugin, but i end up on this error:

[ERROR] Failed to execute goal on project InvUnload: Could not resolve dependencies for project de.jeff_media:InvUnload:jar:4.9.2: The following artifacts could not be resolved: com.plotsquared:PlotSquared:jar:5.11.1, me.clip:placeholderapi:jar:2.10.4: Failure to find com.plotsquared:PlotSquared:jar:5.11.1 in https://hub.spigotmc.org/nexus/content/repositories/snapshots/ was cached in the local repository, resolution will not be reattempted until the update interval of spigot-repo has elapsed or updates are forced -> [Help 1]

It seems like it's expecting PlotSquared 5.11.1 which costs 15 euros which is a bit too much for solving a bug. Maybe i'm doing something wrong that it's expecting PlotSquared 5.11.1 instead of 4? I made a lib directory in the same directory where pom.xml and src is, and put there the 2 dependencies into. SpartanAPI seems to be detected correctly though.

In this video you can see the issue. I first dumpinv some sugarcane, then i do /search dirt, and it shows the laser lines to the chest where i put the sugar cane into. Possibly this helps with reproducing the issue? https://www.youtube.com/watch?v=dS710GZS2E8

Thanks for your time!

**Comment by mfnalex** [comment]
Actually the PlotSquared API used to be free. You can also just compile PS5 yourself: https://github.com/IntellectualSites/PlotSquared/ or just download the API here: https://easyupload.io/iwr6e8 (extract this into your ~/.m2 folder)

**Comment by Engineer1986** [comment]
I finally managed to compile InvUnloadPlus. For future reference i'll note here what i had to do:

Get PlaceholderAPI-2.10.4.jar from https://www.spigotmc.org/resources/placeholderapi.6245/ and put it in C:\Users\ username here \ .m2\repository\me\clip\placeholderapi\2.10.4

Open pom.xml and remove the following on line 153:
```
		<dependency>
			<groupId>com.plotsquared</groupId>
			<artifactId>PlotSquared</artifactId>
			<version>5.11.1</version>
			<scope>provided</scope>
		</dependency>
```

The PlotSquared API only provides PlotSquared-Bukkit and PlotSquared-Core but not the plain PlotSquared. But as it compiled without the reference to PlotSquared (non-bukkit/core), it's apparently fine without it.

I'll look later today if it actually runs in Spigot and then fixing the bug.

**Comment by Engineer1986** [comment]
@mfnalex I have fixed the issue. You can find the new jar, the corrected source file, and before/after screenshots in https://github.com/Engineer1986/tmp

**Comment by mfnalex** [comment]
fixed in 4.9.3
